substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3029,6 +3061,62 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	fn pool_pending_rewards(pool: PoolId) -> Result<BalanceOf<T>, sp_runtime::DispatchError> {
+		let bonded_pool = BondedPools::<T>::get(pool).ok_or(Error::<T>::PoolNotFound)?;
+		let reward_pool = RewardPools::<T>::get(pool).ok_or(Error::<T>::PoolNotFound)?;
+
+		let current_rc = if !bonded_pool.points.is_zero() {
+			let commission = bonded_pool.commission.current();
+			reward_pool.current_reward_counter(pool, bonded_pool.points, commission)?.0
+		} else {
+			Default::default()
+		};
+
+		Ok(PoolMembers::<T>::iter()
```

Comment:
We can't really do this on-chain since this is an unbounded number. Not sure what's the alternative? Some way to page it.

Alternative: We can let any caller just top up any amount and if it is greater than deficit, those tokens will be stuck in reward pool forever and would be equivalent to burning money. But they have no incentive to do it and kind of similar to just transferring tokens to a randomly generated public address.

Reply:
yeah this is 100% a non-starter.

What do you think about the suggestion [here](https://github.com/paritytech/polkadot-sdk/issues/158#issuecomment-1691681473)? Namely: 

> If the pool operator can afford the new ED without their account being destroyed, they should pay the deficit.
> If not, the origin who called the transaction will pay the deficit.

So anyone can force the pool operators to payup. I think this makes a lof sense. Also note that once the pool is destroyed, the deposit goes back to the operator, so it is NOT equivalent to burning.

Reply:
If we are not looking how much is the deficit (since we cannot compute this on-chain), we don't know how much deficit to charge. We cannot then make this permissionless as someone can drain the pool operator's account. I think we can allow only pool operator to top up but then pool would have a deficit until the pool operator fixes it.